### PR TITLE
fix(table-grid, table-semantic) stacked context issue

### PR DESCRIPTION
**Pull request by Deliaz**
#### Description
This PR contains a fix for the stacking context issue, happening with components TableGrid and TableSemantic.
Issue screenshots: 

**Semantic**
Screenshot 2020-10-01 at 12 38 53

**Grid**
Screenshot 2020-10-01 at 12 40 23

**Cause**
The styled HeadCell component in both tables uses position: stacked. Combining with that, if any cell from the table body uses opacity/filter/positioning in it causes the change in stacked context.

**Justification**
Setting zIndex: 1 fixes the issue, making the header cell laying on top of any re-stacked body cells.

#### Scope
Patch: Bug Fix

### CODE REVIEW

**Comment by Deliaz** [comment]
This build-based example: 
https://codesandbox.io/s/basic-usage-forked-voohy

**Comment by Deliaz** [comment]
@chasestarr sure, done.

**Review by sandgraham — COMMENTED** [review]
May want to wrap the entire component in a new stacking context if adding an explicit zindex. Otherwise it may get hidden if someone has their own zindex applied as an ancestor to this component.

**Comment by Deliaz** [comment]
Thanks, @sandgraham  for the review! Do you want me to update PR with that change?

**Comment by chasestarr** [comment]
> Thanks, @sandgraham for the review! Do you want me to update PR with that change?

Do you mind trying to reproduce Graham's concern with a new scenario?

**Comment by Deliaz** [comment]
@chasestarr of course, 

The Graham's assumption is correct, I've compiled a sandbox with demo issue and proposed solution: 
https://codesandbox.io/s/zindex-issue-kfuf9?file=/example.js

Btw, the same problem happens with the current version of Baseweb: 
Screenshot 2020-10-02 at 11 06 08

**Comment by Deliaz** [comment]
@sandgraham I've updated the PR

**Comment by Deliaz** [comment]
Sorry, miss clicked to merge the branch. 
I've noticed that tests are failed. As I ran locally, everything was fine. Is that just a flaky test?

**Comment by chasestarr** [comment]
Yeah likely a flakey puppeteer test. I've restarted CI and plan to merge today

**Review by chasestarr — APPROVED** [review]

File: src/table-grid/styled-components.js [hunk 1]
```diff
@@ -22,6 +22,9 @@ const StyledTableElement = withStyle<
     display: 'grid',
     gridTemplateColumns: props.$gridTemplateColumns,
     flexDirection: 'unset',
+    // Creates a stacking context so we can use z-index on the StyledHeadCell
+    // without affecting anything outside of this component.
+    transform: 'scale(1)',
```
Comment (sandgraham) [inline src/table-grid/styled-components.js:27]: @Deliaz - Sorry I was OOO - this is exactly what I meant - thanks for adding.

**Pull request merged**